pkg/controller/velerobackup/syncset_test.go

```diff
@@ -0,0 +1,280 @@
+package velerobackup
+
+import (
+	"context"
+	"testing"
+
+	"github.com/openshift/hive/pkg/apis"
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1alpha1"
+	controllerutils "github.com/openshift/hive/pkg/controller/utils"
+	testsyncset "github.com/openshift/hive/pkg/test/syncset"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func unchangedSyncSetBase() testsyncset.Option {
+	return func(syncSet *hivev1.SyncSet) {
+		testsyncset.WithName("somesyncset-unchanged")(syncSet)
+		testsyncset.WithNamespace(namespace)(syncSet)
+		testsyncset.WithBackupChecksum(defaultSyncSetChecksumFunc)(syncSet)
+	}
+}
+
+func changedSyncSetBase() testsyncset.Option {
+	return func(syncSet *hivev1.SyncSet) {
+		testsyncset.WithName("somesyncset-changed")(syncSet)
+		testsyncset.WithNamespace(namespace)(syncSet)
+	}
+}
+
+func changedSyncSetReconcileRequests() []reconcile.Request {
+	return []reconcile.Request{
+		{
+			NamespacedName: types.NamespacedName{
+				Namespace: namespace,
+			},
+		},
+	}
+}
+
+var (
+	errSyncSetStatus = errors.StatusError{
+		ErrStatus: metav1.Status{
+			Message: "syncsets.hive.openshift.io \"somesyncset-changed-1\" not found",
+		},
+	}
+
+	defaultSyncSetChecksumFunc = emptyReconcileBackup().syncSetChecksum
+)
+
+func TestSyncSetWatchHandler(t *testing.T) {
+	tests := []struct {
+		name             string
+		mapObject        handler.MapObject
+		expectedRequests []reconcile.Request
+	}{
+		{
+			name: "Not a SyncSet passed in case",
+			mapObject: handler.MapObject{
+				Object: &hivev1.ClusterDeployment{},
+			},
+			expectedRequests: []reconcile.Request{},
+		},
+		{
+			name: "SyncSet unchanged since last backup case",
+			mapObject: handler.MapObject{
+				Object: testsyncset.BuildSyncSet(unchangedSyncSetBase()),
+			},
+			expectedRequests: changedSyncSetReconcileRequests(),
```

Comment:
the function name changedSyncSetReconcileRequest() can probably be improved since it's the same for the changed an unchanged case